src/utils/string.cpp

```diff
@@ -291,6 +291,17 @@ namespace string_util {
     return floating_point(kibibytes / 1024.0 / 1024.0, precision, true, locale) + " GiB";
   }
 
+  /**
+   * Create a GiB string, if the value in GiB is >= 1.0. Otherwise, create a MiB string.
+   */
+  string filesize_gib_mib(unsigned long long kibibytes, size_t precision, const string& locale) {
```

Comment:
I don't think when using MB, it should display anything after the decimal point (just as it already does with the `%mb_*%` tokens). Either have two `precision` arguments, one for MB and one for GB or just always use `0` for the precision when using MB.